File: packages/typia/src/transformers/NodeTransformer.ts
import ts from "typescript";

import { CallExpressionTransformer } from "./CallExpressionTransformer";
import { ITypiaContext } from "./ITypiaContext";

export namespace NodeTransformer {
  export const transform = (props: {
    context: ITypiaContext;
    node: ts.Node;
  }): ts.Node | null =>
    ts.isCallExpression(props.node) && props.node.parent
      ? CallExpressionTransformer.transform({
          context: props.context,
          expression: props.node,
        })
      : props.node;
}

File: packages/typia/native/transform/NodeTransformer.go
package transform

import (
  shimast "github.com/microsoft/typescript-go/shim/ast"
  nativecontext "github.com/samchon/typia/packages/typia/native/core/context"
)

type nodeTransformerNamespace struct{}

var NodeTransformer = nodeTransformerNamespace{}

type NodeTransformer_TransformProps struct {
  Context nativecontext.ITypiaContext
  Node    *shimast.Node
}

func (nodeTransformerNamespace) Transform(props NodeTransformer_TransformProps) *shimast.Node {
  if props.Node != nil && props.Node.Kind == shimast.KindCallExpression && props.Node.Parent != nil {
    return CallExpressionTransformer.Transform(CallExpressionTransformer_TransformProps{
      Context:    props.Context,
      Expression: props.Node.AsCallExpression(),
    })
  }
  return props.Node
}
